csm_web/scheduler/models.py

```diff
@@ -101,6 +109,22 @@ class Student(Profile):
     section = models.ForeignKey("Section", on_delete=models.CASCADE, related_name="students")
     active = models.BooleanField(default=True, help_text="An inactive student is a dropped student.")
 
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        """
+        Create an attendance for this week for the student if their section hasn't already been held this week
+        and no attendance for this week already exists.
+        """
+        now = timezone.now()
+        this_week = week_bounds(now.date())
+        section_day_num = Spacetime.DayOfWeek.values.index(self.section.spacetime.day_of_week)
+        section_already_held = section_day_num < now.weekday() or (
+            section_day_num == now.weekday() and self.section.spacetime.start_time < now.time())
+        course = self.section.course
+        if self.active and course.section_start <= now.date() < course.valid_until\
+                and not section_already_held and not self.attendance_set.filter(date__range=this_week).exists():
+            Attendance.objects.create(student=self, date=now.date())
```

Comment:
Would be useful to add a log message here in case we need to debug in production

Reply:
Added